openmp/libomptarget/plugins-nextgen/common/src/Utils/ELF.cpp

```diff
@@ -283,3 +297,45 @@ Expected<const void *> utils::elf::getSymbolAddress(
 
   return ELFFile.base() + Offset;
 }
+
+template <class ELFT>
+static Expected<StringRef>
+findSymbolInImageImpl(const object::ELFObjectFile<ELFT> &ELFObj,
+                      StringRef Name) {
+  // Search for the symbol by name.
+  auto SymOrErr = getSymbol(ELFObj, Name);
+  if (!SymOrErr)
+    return SymOrErr.takeError();
+  // If symbol not found, return an empty StringRef.
+  if (!*SymOrErr)
+    return StringRef();
```

Comment:
Didn't we used to have a separate boolean check for this? I suppose it works if we want to encode that error logic at the call site.